frappe/commands/site.py

```diff
@@ -13,6 +13,8 @@
 @click.argument('site')
 @click.option('--db-name', help='Database name')
 @click.option('--db-type', default='mariadb', type=click.Choice(['mariadb', 'postgres']), help='Optional "postgres" or "mariadb". Default is "mariadb"')
+@click.option('--db-host', help='Database Host')
+@click.option('--db-port', help='Database Port')
```

Comment:
lets strictly limit port to `int`

```suggestion
@click.option('--db-port', type=int, help='Database Port')
```

Reply:
this also fixes the issue where trying to setup a new-site with just the port specified results in the failure that @surajshetty3416 commented previously.

to replicate, run the following command with no changes to this PR:

```bash
$ bench new-site test.site --db-port 3306
```

the port gets saved as a string in the site_config, causing mysql to fail.